### ISSUE TITLE
C# export passing bad parameters to dotnet publish

### ISSUE THREAD

**Issue by 18swenskiq** [godotengine/godot#73483]
### Godot version

4.0.0-rc.1

### System information

Windows 10

### Issue description
I am trying to export my project (on 4.0.0-rc.1 mono) from the command line. The call looks like this:
godot --path ./game --export-debug "Windows Desktop" "../dist/windows/ultracop.exe"

In the process of calling the dotnet publish command from the `GodotTools.Build` area, the code that it produces will look something like this:
```
dotnet publish H:/issue_recreation/gd-mono-thirdpersoncontroller.csproj -c ExportDebug -r win-x64 --self-contained true -v normal -l:GodotTools.BuildLogger.GodotBuildLogger,E:GodotSource/bin/GodotSharp/Tools/GodotTools.BuildLogger.dll;C:/Users/Quinton/AppData/Roaming/Godot/mono/build_logs\\cea61efc42ba662b94fc2d8419dbd5cd_ExportDebug -p:GodotTargetPlatform=windows -p:GodotRealTIsDouble=true -o C:\\Users\\Quinton\\AppData\\Local\\Temp\\godot-publish-dotnet\\24628-ExportDebug-win-x64
```

I'm unsure what exactly is wrong with it, but dotnet really does not like how the -l parameter is formatted. On my system, it throws this exception:
```
C:/Users/Quinton/AppData/Roaming/Godot/mono/build_logs\\cea61efc42ba662b94fc2d8419dbd5cd_ExportDebug : The term 'C:/Users/Quinton/AppData/Roaming/Godot/mono/build_logs\\cea61efc42ba662b94fc2d8419dbd5cd_ExportDebug' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:236
+ ... dLogger.dll;C:/Users/Quinton/AppData/Roaming/Godot/mono/build_logs\\c ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:/Users/Quinto...5cd_ExportDebug:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

### Steps to reproduce

I am not exactly sure if this is some issue with my system or there is some bigger issue here. The paths all look okay to me and it looks like it's calling the parameter okay. I *have* however found that if dotnet publish is instead called with quotations around the -l parameter, it works fine and I've verified that it actually passes the parameter correctly to dotnet publish.

In this case, the -l parameter is now:
```
"-l:GodotTools.BuildLogger.GodotBuildLogger,E:GodotSource/bin/GodotSharp/Tools/GodotTools.BuildLogger.dll;C:/Users/Quinton/AppData/Roaming/Godot/mono/build_logs\\cea61efc42ba662b94fc2d8419dbd5cd_ExportDebug"
```

This solves the problem, and turning up the verbosity to diagnostic shows that it infact is passing it correctly
```
C:\Program Files\dotnet\sdk\7.0.102\MSBuild.dll --property:_IsPublishing=true --property:GodotTargetPlatform=windows --property:GodotRealTIsDouble=true -property:PublishDir=C:\Users\Quinton\AppData\Local\Temp\godot-publish-dotnet\24628-ExportDebug-win-x64 -property:SelfContained=True -property:_CommandLineDefinedSelfContained=true -property:RuntimeIdentifier=win-x64 -property:_CommandLineDefinedRuntimeIdentifier=true -property:Configuration=ExportDebug -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,C:\Program Files\dotnet\sdk\7.0.102\dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,C:\Program Files\dotnet\sdk\7.0.102\dotnet.dll -l:GodotTools.BuildLogger.GodotBuildLogger,E:GodotSource/bin/GodotSharp/Tools/GodotTools.BuildLogger.dll;C:/Users/Quinton/AppData/Roaming/Godot/mono/build_logs\\cea61efc42ba662b94fc2d8419dbd5cd_ExportDebug -maxcpucount -restore -target:Publish -verbosity:m -verbosity:diagnostic H:/issue_recreation/gd-mono-thirdpersoncontroller.csproj
```

I have actually created a commit for this on my own fork as a proof-of-concept, here: https://github.com/godotengine/godot/compare/master...18swenskiq:godot:master

If people think this is a good fix for this problem, I can submit a pull request

### Minimal reproduction project

I don't think this issue has anything to do with the project itself, as it happens on multiple separate projects on my PC. My guess is that this is either an issue with my PC/paths or a greater issue. I have however included a zip for a project that does cause this issue on my PC though

[issue_recreation.zip](https://github.com/godotengine/godot/files/10763512/issue_recreation.zip)

**Comment by raulsntos** [comment]
We use [`ProcessStartInfo.ArgumentList`](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist) which claims:

> Strings added to the list don't need to be previously escaped.

So I don't understand why we need to add quotes to the parameter.

You mentioned you are exporting the project from the command line, are you using CMD or PowerShell or something else? Also, have you tried exporting from the editor?

**Comment by 18swenskiq** [comment]
So I think I figured out what was going on and I believe it to have been an error on my part. Early on I was trying to pass an absolute path for the export path over the command line which is obviously wrong. For some reason I didn't catch that in the docs and giving it an absolute path makes it throw *very* weird errors. I had fixed that like halfway through debugging this and assumed it was unrelated. I cannot recreate this issue now after passing a relative path on any project and am thus forced to conclude that was the issue.
